File: benchmarks/fair/recursion/mutual.c
#include <stdio.h>
#include <stdlib.h>
static long long odd_(long long n);
static long long even_(long long n){ return n==0?1:odd_(n-1); }
static long long odd_(long long n){ return n==0?0:even_(n-1); }
int main(void){const char*e=getenv("BENCH_N");long long n=e?atoll(e):0;if(n<=0)n=30000000;
    long long s=0; for(long long i=0;i<n;i+=1000) s+=even_(i%1000);
    printf("%lld\n",s);return 0;}

File: benchmarks/fair/recursion/mutual.rs
fn even_(n:i64)->i64{ if n==0 {1} else {odd_(n-1)} }
fn odd_(n:i64)->i64{ if n==0 {0} else {even_(n-1)} }
fn main(){let n:i64=std::env::var("BENCH_N").ok().and_then(|v|v.parse().ok()).filter(|&v|v>0).unwrap_or(20000);
    let mut s=0i64; for i in 0..n { s+=even_(i%997); } println!("{}",s);}
